access/validator.go

```diff
@@ -159,6 +183,14 @@ func (v *TransactionValidator) Validate(tx *flow.TransactionBody) (err error) {
 		return err
 	}
 
+	err = v.checkSufficientBalanceToPayForTransaction(ctx, tx)
+	if err != nil {
+		var balanceError InsufficientBalanceError
+		if errors.As(err, &balanceError) {
```

Comment:
a pattern we use frequently is to add a method `IsInsufficientBalanceError` that wraps this logic into a function.

Reply:
can you add a comment here explaining why this check is needed, and why we do not return an error in the else case?

Reply:
we should log the error in the else case instead of silently ignoring them